src/core/protocol-authorization.ts

```diff
@@ -152,6 +153,47 @@ export class ProtocolAuthorization {
     );
   }
 
+  public static async authorizeSubscription(
```

Comment:
Design discussion (I am sure you anticipated some form of discussion): this method (as well as the `authorizeSubscribe()` in `RecordsGrantAuthorization`) appear to be entirely the same, minus the type info. Given this is not a coincidence, ie. you intended for the auth to be identical, it seems like it's worth it to not duplicate code to realize that intent. This probably involves generalizing the type differences, with the hope that this doesn't involve crazy gymnastics...

Reply:
Let's chat about this so that I make sure I'm following what you mean. I think I do.

Reply:
Done!

There's room for further removal of duplication as we talked about, but likely would be better in a separate effort.